benchopt/utils/github.py

```diff
@@ -24,8 +25,12 @@ def publish_result_file(benchmark_name, file_path, token):
         raise FileNotFoundError(
             f"Could not upload file {file_to_upload}."
         )
-    with file_to_upload.open('r') as f:
-        file_content = f.read()
+
+    if file_to_upload.suffix == ".csv":
+        with file_to_upload.open('r') as f:
```

Comment:
to have a binary file you would do:
```suggestion
        with file_to_upload.open('rb') as f:
```

Reply:
Thanks